src/Encoder.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef NBS_ENCODER_HPP
+#define NBS_ENCODER_HPP
+
+#include <fstream>
+#include <napi.h>
+
+#include "Packet.hpp"
+#include "zstr/zstr.hpp"
+
+namespace nbs {
+    class Encoder : public Napi::ObjectWrap<Encoder> {
+    public:
+        /// Initialize Encoder class NAPI binding
+        static Napi::Object Init(Napi::Env& env, Napi::Object& exports);
+
+        Encoder(const Napi::CallbackInfo& info);
```

Comment:
Add docstrings for each of these in the header (and for all the other functions added in headers)